src/main/java/com/ksm/robolo/roboloapp/restsecurity/config/WebSecurityConfig.java

```diff
@@ -0,0 +1,86 @@
+package com.ksm.robolo.roboloapp.restsecurity.config;
+
+import com.ksm.robolo.roboloapp.restsecurity.RESTAuthenticationEntryPoint;
+import com.ksm.robolo.roboloapp.restsecurity.RESTAuthenticationFailureHandler;
+import com.ksm.robolo.roboloapp.restsecurity.RESTAuthenticationSuccessHandler;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.authentication.dao.DaoAuthenticationProvider;
+import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.builders.WebSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+@EnableWebSecurity
+@Configuration
+public class WebSecurityConfig extends WebSecurityConfigurerAdapter{
+
+    private final UserDetailsService userDetailsService;
+
+    private final PasswordEncoder passwordEncoder;
+
+    private final RESTAuthenticationEntryPoint authenticationEntryPoint;
+
+    private final RESTAuthenticationFailureHandler authenticationFailureHandler;
+
+    private final RESTAuthenticationSuccessHandler authenticationSuccessHandler;
+
+    @Autowired
```

Comment:
Trochę off review, ale jestem ciekaw jak w tym przypadku zadziała ta adnotacja

Reply:
normalnie, to jest wstrzykiwanie przez konstruktor, a nie przez pole ;) to robi dokładnie tak samo, ale tak jest uważane za lepszą praktykę, ponieważ nie ukrywasz wtedy dependencji klasy. Plus jest jeszcze coś takiego, że chyba koniec końców spring i tak to musi sobie jakoś przerobić na konstruktor. Jakaś ReflectionAPI magia się dzieje.